Platforms+obstacles+lvl's

Aramymo opened the pull request.

Aramymo committed:
Bug Fix

Fixed a Bug when two players were appearing on the screen

Aramymo committed:
Life Model, Enemy Logic, New Spawn Logic (raw)

Added life model with basic stats, combat system and a new way to spawn the player in the levels

HozzyUwU commented on Assets/Scripts/CombatHandler.cs:
```diff
@@ -0,0 +1,47 @@
+    {
+        if (other.CompareTag("Enemy"))
+        {
+            _playerController._isInCombat = true;
```
Переменные типа public не должны начинаться с "_", сделай ее private, а взаимодействие реализуй через [properties](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties).

HozzyUwU commented on Assets/Scripts/EnemyScripts/EnemyLife.cs:
```diff
@@ -0,0 +1,29 @@
+        _attackSpeed = 2.0f;
+    }
+
+    public void Die()
+    {
+        Debug.Log("Enemy is dead!");
+        Destroy(this);
+    }
```
Не стоит плодить одинаковые методы в разных классах, сделай его виртуальным/абстрактным в Stats.

HozzyUwU commented on Assets/Scripts/EnemyScripts/EnemyLife.cs:
```diff
@@ -0,0 +1,29 @@
+        Destroy(this);
+    }
+
+    void Update()
+    {
+        //Debug.Log("Enemy health: " + _health);
+        if (_health <= 0)
+        {
+            Die();
+        }
+    }
```
Нет нужды делать подобную проверку каждый кадр, у тебя весь урон проходит через один единственный метод, делай проверку в нем(Stats).

HozzyUwU commented on Assets/Scripts/Stats.cs:
```diff
@@ -0,0 +1,40 @@
+        //Debug.Log(_health);
+    }
+
+    public float DoDamage()
+    {
+        if(Time.time >= _nextDamageEvent)
+        {
+            _nextDamageEvent = Time.time + _attackSpeed;
+            Debug.Log("HITHITHIT");
+            return _strength;
+        }
+        else
+        {
+            //_nextDamageEvent = Time.time + _attackSpeed;
+            Debug.Log("Waiting");
+            return 0.0f;
+        }
+    }
```
Точно не стоит логику со скоростью нанесения атак обрабатывать в этом классе. Оставь здесь только значение урона, а всю обработку перенеси в CombatHandler.

HozzyUwU reviewed (changes requested):
В целом работает, но есть некоторые замечания. По возможности прошу разобраться с организацией файловой системы, лично мне весьма трудно ориентироваться в пространстве.

Aramymo committed:
Bug fixes

Fixed all bugs in code review + in testing task #2

Aramymo committed:
Inventory system v0.5

Added inventory system and items, player saves item on impact in his inventroy, where he can select an item and see its desription.
